### PR TITLE
allow Dates to be numerically sorted by SortIterator

### PR DESCRIPTION
**Pull request by antograssiot**
Solves #7095 by converting automatically DateTime to seconds when SORT_NUMERIC is used.

### CODE REVIEW

**Comment by markstory** [comment]
Looks good to me.

File: src/Collection/Iterator/SortIterator.php [hunk 1]
```diff
@@ -67,6 +67,9 @@ public function __construct($items, $callback, $dir = SORT_DESC, $type = SORT_NU
         $callback = $this->_propertyExtractor($callback);
         $results = [];
         foreach ($items as $key => $value) {
+            if ($value instanceof \DateTime && $type === SORT_NUMERIC) {
+                $value = $value->format('U');
+            }
             $results[$key] = $callback($value);
         }
```
Comment (lorenzo) [inline src/Collection/Iterator/SortIterator.php]: The type check should be done after $callback($value)

**Comment by antograssiot** [comment]
@lorenzo I've updated the test case also, I hope it will fit your requirements.

**Comment by markstory** [comment]
Looks good.

**Comment by antograssiot** [comment]
@lorenzo I updated as you requested on IRC. Should be good to go now

**Comment by lorenzo** [comment]
:clap:

**Comment by kyleweishaupt** [comment]
Thanks everyone!

**Pull request merged**